src/main/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/producer/AvroSerializer.java

```diff
@@ -0,0 +1,40 @@
+package uk.gov.companieshouse.chipsrestinterfacesconsumer.producer;
+
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.generic.GenericDatumWriter;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.io.BinaryEncoder;
+import org.apache.avro.io.EncoderFactory;
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.chips.ChipsRestInterfacesSend;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+
+@Component
+public class AvroSerializer {
+
+    public byte[] serialize(ChipsRestInterfacesSend message, Schema schema) throws IOException {
+        GenericDatumWriter<GenericRecord> datumWriter = new GenericDatumWriter<>(schema);
+        ByteArrayOutputStream stream = new ByteArrayOutputStream();
+        BinaryEncoder encoder = EncoderFactory.get().binaryEncoder(stream, null);
+        datumWriter.write(buildAvroGenericRecord(message, schema), encoder);
+        encoder.flush();
+        byte[] dataValue = stream.toByteArray();
+        stream.close();
+        return dataValue;
+    }
+
+    private GenericRecord buildAvroGenericRecord(ChipsRestInterfacesSend message, Schema schema) {
```

Comment:
we don't need to convert it to a generic record anymore, the kafka model can be serialized directly, it's what i'm doing in the api (next PR) - see here for an example - https://github.com/companieshouse/kafka-models#using-the-model-for-serialising-with-avro

Reply:
Removed generic record.